### PR TITLE
RPC: Get rid of the internal miner's hashmeter

### PR DESCRIPTION
**Pull request by jtimon**
Removes "hashespersec" in "getmininginfo" and also removes "gethashespersec".
That also allows the removal of some code in miner.cpp, simplifying ScanHash() and BitcoinMiner().

### CODE REVIEW

**Comment by gavinandresen** [comment]
Code review ACK, but there is a functional change: with this code the internal miner will never search the nonce space above 0xfff.

Also: how did you test this?

**Comment by jtimon** [comment]
We discussed this in #4793 . @sipa said in some outdated commit that removing ```if ((nNonce & 0xffff) == 0)``` was safe but I can't remember the reason why it was that if instead of the later if ((nNonce & 0xfff) == 0) what can be removed.
Now that you're saying it I think it is a mistake and what can be removed is just:
```
if ((nNonce & 0xfff) == 0)
            boost::this_thread::interruption_point();
```

**Comment by sipa** [comment]
@gavinandresen I haven't tested, but it's just ScanHash that returns after 0x1000 iterations - the mining loop continues with the same nonce until 0xffff0000 is reached (which could actually become 0xfffff000 then, but who cates).

**Comment by sipa** [comment]
Tested ACK: adding a `LogPrintf("Nonce: %x\n", nNonce);` in the loop that calls ScanHash, and running `setgenerate true` in `-testnet` mode confirms higher nonces are being tried.

**Comment by jtimon** [comment]
Oh, here it was the missing piece of earlier conversation: https://github.com/jtimon/bitcoin/commit/9d12ff72a6f97ed1e53e0d106abe1a3a7e1f56f8#diff-4a59b408ad3778278c3aeffa7da33c3cR384

**Comment by laanwj** [comment]
@luke-jr this is mining related, can you ACK?

File: src/miner.cpp [hunk 1]
```diff
@@ -395,10 +393,8 @@ bool static ScanHash(const CBlockHeader *pblock, uint32_t& nNonce, uint256 *phas
             return true;
 
         // If nothing found after trying for a while, return -1
-        if ((nNonce & 0xffff) == 0)
-            return false;
         if ((nNonce & 0xfff) == 0)
-            boost::this_thread::interruption_point();
+            return false;
```
Comment (luke-jr) [inline src/miner.cpp]: I think this is okay, but seems unrelated from the PR description?

**Comment by luke-jr** [comment]
ACK, only tested on regtest since testnet-in-a-box style apparently refuses to mine in master... :/

**Comment by luke-jr** [comment]
ACK on testnet via gdb testing

File: src/miner.cpp (hunk 1 above)
Comment (jtimon) [inline src/miner.cpp]: @maaku suggested on #4793 when I changed this ```boost::this_thread::interruption_point()``` to ```return false``` that the change belonged to the removal of the hashmeter as it was the reason for the interruption point to be there in the first place.

**Comment by gavinandresen** [comment]
Better is better... so ACK.

But after removing the hashmeter, and assuming we no longer care about performance of the CPU miner (we don't) it looks to me like the code could be simplified a lot by getting rid of ScanHash and just having the mining loop be:
```
   while true:
       ... create block...
       ... serialize header without nonce..
       set nonce to zero
       do:
           compute hash with nonce
           ++nonce
           if hash <= target: BLOCK FOUND
           if nonce%0xfff : interruption point
        while nonce < 0xffffffff
        Increment extraNonce
  end while true
```

**Comment by jtimon** [comment]
Yes, more simplifications can be done later.
This is what I had in mind for next steps: https://github.com/jtimon/bitcoin/commit/a029a67fd29bcc04b463bd4dcf7507b5b7e0ed7f
https://github.com/jtimon/bitcoin/commit/da36fb5ebb8655392ec278dd632bdf7a52531e0f
Different things can be done (my preference is that we share as much code as possible with regtest, optimizing only for regtest), but most of the options I have considered imply the removal of the hashmeter as a first step, so here it is isolated from other decisions that people seem to care less about.

**Comment by jaromil** [comment]
ut ACK.

lovely to have this part of the code simplified.
